### ISSUE TITLE
eval_mle over binary_field_ghash SIGSEGVs on the CUDA backend (mle_coeffs_to_evals OK)

### ISSUE THREAD

**Issue by chokobole** [fractalyze/zorch#400]
## Goal

`zorch.poly.multilinear.eval_mle` should evaluate a multilinear over
`binary_field_ghash` on the CUDA backend without crashing, the way it already
does for prime / extension fields (and the way `mle_coeffs_to_evals` already does
for `binary_field_ghash`).

## Symptom

On the CUDA backend, `eval_mle` over `binary_field_ghash` **hard-SIGSEGVs**
(process exits 139, no Python traceback, core dump). `mle_coeffs_to_evals` over
the same dtype runs fine — only `eval_mle`'s fold crashes.

Minimal repro (`JAX_PLATFORMS=cuda JAX_ENABLE_X64=1`, zorch on `PYTHONPATH`):

```python
import jax, jax.numpy as jnp
from jax import lax
jax.config.update("jax_enable_x64", True)
BFG = jnp.dtype("binary_field_ghash")
from zorch.poly.multilinear import eval_mle, mle_coeffs_to_evals

# 2-var multilinear; values are F128 (lo, hi) pairs bitcast to ghash.
coeffs = lax.bitcast_convert_type(jnp.asarray([[1,0],[2,0],[3,0],[4,0]], jnp.uint64), BFG)
point  = lax.bitcast_convert_type(jnp.asarray([[5,0],[6,0]], jnp.uint64), BFG)

evals = mle_coeffs_to_evals(coeffs); evals.block_until_ready()   # OK
v = eval_mle(evals, point); v.block_until_ready()                # SIGSEGV (exit 139)
```

Observed: `mle_coeffs_to_evals OK` prints, then the process dies on `eval_mle`.

## Why

`eval_mle` is on the critical path of the **code-generic Ligerito round-trip**
(zorch#378): the value check `claim == Σ_x residual(x)·B(x)` and the induced
proximity basis both call it. flock-zorch#32 T4 needs that round-trip to run over
`binary_field_ghash` on GPU; today it cannot.

The commitment layer is unaffected — the Ligero/Ligerito matrix commit is
`ReedSolomon.encode` (additive NTT) + Merkle, neither of which touches
`eval_mle`. flock-zorch#32's M0 commitment byte-match against flock-core is
already green on GPU; this bug is the gate on the recursive-driver half.

Sibling of zorch#381 (both are binary-field GPU-lowering gaps on the multilinear
surface). CPU is not a fallback here: binary-field arithmetic is unlowered on the
CPU PJRT path in the flock-zorch env (`a*b` returns 0; scatter fails to legalize
`i128 -> !field.bf<7, ghash>`).

## End criteria

- The repro above runs to completion on CUDA and prints a non-crashing
  `eval_mle` result.
- `eval_mle` over `binary_field_ghash` is covered by a GPU test (sibling of
  whatever pins `mle_coeffs_to_evals`).
- flock-zorch#32's code-generic Ligerito round-trip can be exercised over the
  GHASH instantiation on GPU.

**Comment by chokobole** [comment]
Root-caused — this is **not a zorch bug**. The `eval_mle` SIGSEGV (and every field reduction) is a **compile-time crash in the XLA GPU emitter**: `RewriteShuffleReduce` (`LowerXlaToScfPass`, `xla/codegen/emitters/transforms/lower_xla_to_scf.cc`) lowers a warp-shuffle reduction by bitcasting to storage-int, with branches for prime/extension/EC field types but **not** `BinaryFieldType`. So `!field.bf<7,ghash>` falls through to `getIntOrFloatBitWidth()` — an invalid FloatType downcast on a field type — and segfaults. That's exactly why prime reduce codegens and binary crashes, and why the gap is the whole reduction surface (46 sites), not `eval_mle`.

Fix is in **fractalyze/xla** (add the binary-field branch, mirroring prime/extension; `field.bitcast` already lowers for binary via `ConvertBinaryFieldBitcast`, so no prime-ir change). The zorch-side `field_sum` workaround (#402) is **closed as superseded**. This issue resolves once the emitter fix ships as a jax-cuda12 wheel + the flock pin bumps — then `jnp.sum` works natively over binary fields on GPU and no library shim is needed.

**Comment by chokobole** [comment]
Duplicate of #398 (same SIGSEGV, same repro shape, filed in parallel that evening). Unique context — critical-path note, CPU-not-a-fallback detail, end criteria — ported to #398 in a comment; track there.
